### PR TITLE
[pallet-revive] implement the base fee API

### PR DESCRIPTION
**Pull request by xermicus**
This PR implements the base fee syscall API method. Currently this is implemented as a compile time constant in the revive compiler, returning 0. However, since this is an opocde, if we ever need to implement it for compatibility reasons with [EIP-1559](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md), it would break already deployed contracts. Thus we provide a syscall method instead.

### CODE REVIEW

**Comment by xermicus** [comment]
bot fmt

**Comment by athei** [comment]
> It returns an u64 because ref_time is also u64 and the base fee is always less.

Why? Isn't this a balance value which could theoretically exceed u64?

**Comment by xermicus** [comment]
> > It returns an u64 because ref_time is also u64 and the base fee is always less.
> 
> Why? Isn't this a balance value which could theoretically exceed u64?

Ah yes of course - I just short-circuited it because the gas price is currently `1`. But it might exceed it in theory. I'm going to change it to return a u256.

**Comment by athei** [comment]
> > > It returns an u64 because ref_time is also u64 and the base fee is always less.
> > 
> > 
> > Why? Isn't this a balance value which could theoretically exceed u64?
> 
> Ah yes of course - I just short-circuited it because the gas price is currently `1`. But it might exceed it in theory. I'm going to change it to return a u256.

Strictly speaking it is capped to u128 which is our balance type everywhere. But it is a generic type. And in these cases we exposed it as u256 in all other places.

**Comment by xermicus** [comment]
@athei yes I fully agree.

File: substrate/frame/revive/fixtures/contracts/base_fee.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
```
Comment (jarkkojs) [inline substrate/frame/revive/fixtures/contracts/base_fee.rs:10]: Not demanding to remove this in principle SPDX already fixes this up, and rest is redundant information.

Comment (athei) [reply substrate/frame/revive/fixtures/contracts/base_fee.rs:10]: If we want to change the header it needs to be a separate PR changing headers in all files.

**Review by athei — APPROVED** [review]

**Pull request merged**